tests/test_unittest/test_call_interface.py

```diff
@@ -194,3 +195,172 @@ def test_domain_selection():
+    else:
+        assert False
+
+    arg1 = gt_storage.ones(
+        backend=backend, dtype=np.float64, shape=(3, 3, 3), default_origin=(0, 0, 0)
+    )
+    arg2 = gt_storage.zeros(
+        backend=backend, dtype=np.float64, shape=(3, 3, 3), default_origin=(0, 0, 0)
+    )
+    arg3 = gt_storage.ones(
+        backend=backend, dtype=np.float64, shape=(3, 3, 3), default_origin=(0, 0, 0)
+    )
+    tmp = np.asarray(arg3)
+    tmp *= 2
+
+    branch_true(arg1, arg2=None, par1=2.0, par2=5.0, par3=3.0)
+    np.testing.assert_equal(arg1, 10 * np.ones((3, 3, 3)))
+    branch_true(arg1, arg2=None, par1=2.0, par2=5.0)
+    np.testing.assert_equal(arg1, 100 * np.ones((3, 3, 3)))
+    branch_false(arg1, arg2, arg3, par1=2.0, par2=5.0, par3=3.0)
+    np.testing.assert_equal(arg1, 60 * np.ones((3, 3, 3)))
+    try:
+        branch_false(arg1, arg2, arg3, par1=2.0, par2=5.0)
+    except ValueError:
+        pass
+    else:
+        assert False
+
+
+@pytest.mark.parametrize(
+    "backend",
+    [
+        name
+        for name in gt_backend.REGISTRY.names
+        if gt_backend.from_name(name).storage_info["device"] != "gpu"
+    ],
+)
+def test_halo_checks(backend):
+    stencil = gtscript.stencil(definition=avg_stencil, backend=backend)
+
+    # test default works
+    in_field = gt_storage.ones(
+        backend=backend, shape=(22, 22, 10), default_origin=(1, 1, 0), dtype=np.float64
+    )
+    out_field = gt_storage.zeros(
+        backend=backend, shape=(22, 22, 10), default_origin=(1, 1, 0), dtype=np.float64
+    )
+    stencil(in_field=in_field, out_field=out_field)
+    assert (out_field[1:-1, 1:-1, :] == 1).all()
+
+    # test setting arbitrary, small domain works
+    in_field = gt_storage.ones(
+        backend=backend, shape=(22, 22, 10), default_origin=(1, 1, 0), dtype=np.float64
+    )
+    out_field = gt_storage.zeros(
+        backend=backend, shape=(22, 22, 10), default_origin=(1, 1, 0), dtype=np.float64
+    )
+    stencil(in_field=in_field, out_field=out_field, origin=(2, 2, 0), domain=(10, 10, 10))
+    assert (out_field[2:12, 2:12, :] == 1).all()
+
+    # test setting domain+origin too large raises
+    in_field = gt_storage.ones(
+        backend=backend, shape=(22, 22, 10), default_origin=(1, 1, 0), dtype=np.float64
+    )
+    out_field = gt_storage.zeros(
+        backend=backend, shape=(22, 22, 10), default_origin=(1, 1, 0), dtype=np.float64
+    )
+    with pytest.raises(ValueError):
+        stencil(in_field=in_field, out_field=out_field, origin=(2, 2, 0), domain=(20, 20, 10))
+
+    # test 2*origin+domain does not raise if still fits (c.f. previous bug in c++ check.)
+    in_field = gt_storage.ones(
+        backend=backend, shape=(23, 23, 10), default_origin=(1, 1, 0), dtype=np.float64
+    )
+    out_field = gt_storage.zeros(
+        backend=backend, shape=(23, 23, 10), default_origin=(1, 1, 0), dtype=np.float64
+    )
+    stencil(in_field=in_field, out_field=out_field, origin=(2, 2, 0), domain=(20, 20, 10))
+
+
```

Comment:
A test is missing to check that stencil works if sequences of integral values other than `tuple`s are passed to `origin` and `domain` arguments (spoiler: it doesn't)